mypy-strict.ini

```diff
@@ -0,0 +1,78 @@
+# This is the PyTorch mypy-strict.ini file (note: don't change this line! -
+# test_run_mypy in test/test_type_hints.py uses this string)
+
+# Unlike mypy.ini, it enforces very strict typing rules. The intention is for
+# this config file to be used to ENFORCE that people are using mypy on codegen
+# files.
+
+[mypy]
+python_version = 3.8
+#plugins = mypy_plugins/check_mypy_version.py, numpy.typing.mypy_plugin
```

Comment:
This is the only change I made to these files. I don't think we need the mypy version check since we install 1.4.0 in the lintrunner config anyway. I haven't looked through everything else yet. We might want a minimal set of options, or to disable the strict version.

Reply:
Another simple approach is to remove the strict version and also remove the `--config=mypy.ini` argument (and these files), so that we'd use the default settings. We'd probably at least want to set `--ignore-missing-imports`, so perhaps trimming down `mypy.ini` to just a few options is preferable.

Reply:
I'm wondering if we should just blankly skip checks on specific libraries? i.e.
```
[mypy]
ignore_missing_imports = True
```

I know little about mypy, let me know if I'm talking non-sense :laughing: